test/interpreter/function-arabic.spec.ts

```diff
@@ -0,0 +1,100 @@
+    const engine = HyperFormula.buildFromArray([
+      ['=ARABIC(" XD ")'],
+      ['=ARABIC("xd")'],
+      ['=ARABIC(" xD ")'],
+    ])
+    expect(engine.getCellValue(adr('A1'))).toEqual(490)
+    expect(engine.getCellValue(adr('A2'))).toEqual(490)
+    expect(engine.getCellValue(adr('A3'))).toEqual(490)
+  })
+
+  it('should detect incorrect numerals', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=ARABIC("IMM")'],
+      ['=ARABIC("MMMM")'],
+      ['=ARABIC("IXC")'],
+      ['=ARABIC("--I")'],
+      ['=ARABIC("-")'],
+      ['=ARABIC("Ma")'],
+      ['=ARABIC("M M")'],
+    ])
+    expect(engine.getCellValue(adr('A1'))).toEqualError(detailedError(ErrorType.VALUE, ErrorMessage.Roman))
+    expect(engine.getCellValue(adr('A2'))).toEqualError(detailedError(ErrorType.VALUE, ErrorMessage.Roman))
+    expect(engine.getCellValue(adr('A3'))).toEqualError(detailedError(ErrorType.VALUE, ErrorMessage.Roman))
+    expect(engine.getCellValue(adr('A4'))).toEqualError(detailedError(ErrorType.VALUE, ErrorMessage.Roman))
+    expect(engine.getCellValue(adr('A5'))).toEqualError(detailedError(ErrorType.VALUE, ErrorMessage.Roman))
+    expect(engine.getCellValue(adr('A6'))).toEqualError(detailedError(ErrorType.VALUE, ErrorMessage.Roman))
+    expect(engine.getCellValue(adr('A7'))).toEqualError(detailedError(ErrorType.VALUE, ErrorMessage.Roman))
+  })
+
+  it('works for border cases', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=ARABIC("MMMIMIDCCCICILXXXIXIVIII")'],
+      ['=ARABIC("-I")'],
+    ])
+    expect(engine.getCellValue(adr('A1'))).toEqual(4992)
+    expect(engine.getCellValue(adr('A2'))).toEqual(-1)
+  })
+
+  it('should output correct value for roman numerals from mode 0', () => {
+    const [input, output] = inputOutput(0)
+    const engine = HyperFormula.buildFromArray([input])
+    expect(engine.getSheetValues(0)).toEqual([output])
+  })
+
+  it('should output correct value for roman numerals from mode 1', () => {
+    const [input, output] = inputOutput(1)
+    const engine = HyperFormula.buildFromArray([input])
+    expect(engine.getSheetValues(0)).toEqual([output])
+  })
+
+  it('should output correct value for roman numerals from mode 2', () => {
+    const [input, output] = inputOutput(2)
+    const engine = HyperFormula.buildFromArray([input])
+    expect(engine.getSheetValues(0)).toEqual([output])
+  })
+
+  it('should output correct value for roman numerals from mode 3', () => {
+    const [input, output] = inputOutput(0)
+    const engine = HyperFormula.buildFromArray([input])
+    expect(engine.getSheetValues(0)).toEqual([output])
+  })
+
+  it('should output correct value for roman numerals from mode 4', () => {
+    const [input, output] = inputOutput(4)
+    const engine = HyperFormula.buildFromArray([input])
+    expect(engine.getSheetValues(0)).toEqual([output])
+  })
+})
+
+function inputOutput(mode: number) {
+  const arr = [mode0, mode1, mode2, mode3, mode4][mode]
```

Comment:
We can create this array once, outside of the function. We run tests a lot, maybe it's desired to reduce time and memory footprint. WDYT?

```suggestion
const modes = [mode0, mode1, mode2, mode3, mode4];
function inputOutput(mode: number) {
  const arr = modes[mode]
```

Reply:
there is no memory copying involved in the creation of this array, so this is already fast

Reply:
(and this code is run 5 times in total during testing)

Reply:
Each test adds to the total 20m tests run. But ok, we can leave it as is

Reply:
the speedup is not even measurable, and due to quirks of JS compilation could not even be a speedup